dragonfly/block/kelp.go

```diff
@@ -0,0 +1,144 @@
+package block
+
+import (
+	"github.com/df-mc/dragonfly/dragonfly/entity/physics"
+	"github.com/df-mc/dragonfly/dragonfly/item"
+	"github.com/df-mc/dragonfly/dragonfly/world"
+	"github.com/go-gl/mathgl/mgl64"
+	"math/rand"
+	"time"
+)
+
+// Kelp is an underwater block which can grow on top of solids underwater.
+type Kelp struct {
+
+	// Age is the age of the kelp block which can be 0-15. If age is 15, kelp won't grow any further.
+	Age int
+}
+
+// BreakInfo ...
+func (k Kelp) BreakInfo() BreakInfo {
+	// Kelp can be instantly destroyed.
+	return BreakInfo{
+		Hardness:    0.0,
+		Harvestable: alwaysHarvestable,
+		Effective:   nothingEffective,
+		Drops:       simpleDrops(item.NewStack(k, 1)),
+	}
+}
+
+// EncodeItem ...
+func (Kelp) EncodeItem() (id int32, meta int16) {
+	return 335, 0
+}
+
+// EncodeBlock ...
+func (k Kelp) EncodeBlock() (name string, properties map[string]interface{}) {
+	return "minecraft:kelp", map[string]interface{}{"age": int32(k.Age)}
+}
+
+// CanDisplace ...
+func (Kelp) CanDisplace(b world.Liquid) bool {
+	_, water := b.(Water)
+	return water
+	// Kelp can waterlog.
+}
+
+// SideClosed ...
+func (Kelp) SideClosed(pos, side world.BlockPos, w *world.World) bool {
+	return false
+	// Kelp can always be flowed through.
+}
+
+// AABB ...
+func (Kelp) AABB(world.BlockPos, *world.World) []physics.AABB {
+	return nil
+	// Kelp can be placed even if someone is standing on its placement position.
+}
+
+// SetRandomAge ...
+func (k Kelp) setRandomAge() Kelp {
+	// In Java Edition, the age value can be up to 25, but MCPE limits it to 15.
+	k.Age = rand.Intn(14)
+	return k
+}
+
+// UseOnBlock ...
+func (k Kelp) UseOnBlock(pos world.BlockPos, face world.Face, _ mgl64.Vec3, w *world.World, user item.User, ctx *item.UseContext) (used bool) {
+	pos, _, used = firstReplaceable(w, pos, face, k)
+	if !used {
+		return
+	}
+
+	switch w.Block(pos.Add(world.BlockPos{0, -1})).(type) {
+	// Kelp blocks must be placed on a solid or another kelp block, TODO: Replace this to check for a solid in the future when a Solid interface exists.
+	case Air, Water:
+		return false
+	}
+
+	liquid, liquidExists := w.Liquid(pos)
+	// A check for existent water at this position, kelp cannot be placed if none is found.
+	if !liquidExists || liquid.LiquidType() != "water" || liquid.LiquidDepth() < 8 {
+		// Water must be a source block to plant kelp.
+		return false
+	}
```

Comment:
Could you change this to something like this:
```go
liquid, ok := w.Liquid(pos)
	if !ok { 
		return false
	} else if _, ok := liquid.(Water); ok || liquid.LiquidDepth() < 8 {
		return false
	}
```
Ideally we'd avoid using strings like that, LiquidType() is only really there for checking if two liquids are of the same type.